### ISSUE TITLE
Tests failing for FilterStateChoices - format shortens names if strings are too long

### ISSUE THREAD

**Issue by m7pr** [insightsengineering/teal.slice#306]
### What happened?

Tests failing for FilterStateChoices - format shortens names if strings are too long

```{R}
==> devtools::test()

ℹ Testing teal.slice
✔ | F W S  OK | Context
✔ |         9 | calls_combine_by [0.1s]                                                              
✖ | 1      55 | ChoicesFilterState [0.4s]                                                            
─────────────────────────────────────────────────────────────────────────────────────────────────────
Failure (test-ChoicesFilterState.R:343:3): format shortens names if strings are too long
shiny::isolate(filter_state$format()) not equal to paste(...).
1/1 mismatches
x[1]: "ChoicesFilterState:\nteal_slice\n $ dataname: data\n $ varname : variable\n $ choices : exceed
x[1]: ingly long value nam...\n $ selected: exceedinglylongvaluenameex...\n $ fixed   : FALSE\n $ dis
x[1]: abled: FALSE\n"
y[1]: "ChoicesFilterState:\nteal_slice\n $ dataname: \"data\"\n $ varname : \"variable\"\n $ choices 
y[1]: : c(\"exceedingly long value ...\n $ selected: c(\"exceedinglylongvaluenam...\n $ fixed   : FAL
y[1]: SE\n $ disabled: FALSE\n"
─────────────────────────────────────────────────────────────────────────────────────────────────────
✔ |        50 | count_labels [0.1s]                                                                  
✔ |        37 | DateFilterState [0.2s]                                                               
✔ |        37 | DatetimeFilterState [0.2s]                                                           
✔ |        20 | DefaultFilteredDataset [0.3s]                                                        
✔ |         3 | DFFilterStates                                                                       
✔ |        12 | EmptyFilterState                                                                     
✔ |        23 | filter_panel_api [5.9s]                                                              
✔ |        95 | FilteredData [5.6s]                                                                  
✔ |        16 | FilteredDataset [0.2s]                                                               
✔ |        22 | FilterState-utils                                                                    
✔ |        57 | FilterState [0.4s]                                                                   
✔ |        28 | FilterStateExpr [0.1s]                                                               
✔ |        35 | FilterStates [0.7s]                                                                  
✔ |         5 | get_supported_filter_varnames                                                        
✔ |        37 | init_filter_state [0.1s]                                                             
✔ |         8 | init_filter_states                                                                   
✔ |         8 | init_filtered_data                                                                   
✔ |         4 | init_filtered_dataset [0.2s]                                                         
✔ |        24 | LogicalFilterState [0.1s]                                                            
✔ |        19 | MAEFilteredDataset [3.6s]                                                            
✔ |         4 | MAEFilterStates [0.3s]                                                               
✔ |         5 | MatrixFilterStates                                                                   
✔ |        49 | RangeFilterState [0.5s]                                                              
✔ |        11 | SEFilterStates [0.3s]                                                                
✔ |       133 | teal_slice [0.4s]                                                                    
✔ |        35 | utils [0.2s]                                                                         

══ Results ══════════════════════════════════════════════════════════════════════════════════════════
Duration: 21.6 s

[ FAIL 1 | WARN 0 | SKIP 0 | PASS 841 ]
```

### sessionInfo()

```R
> sessionInfo()
R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.utf8  LC_CTYPE=C                            
[3] LC_MONETARY=English_United States.utf8 LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] scda_0.1.6.9000      teal_0.12.0.9026     teal.transform_0.3.0 magrittr_2.0.3      
[5] teal.data_0.2.0      teal.slice_0.3.0     shiny_1.7.4         

loaded via a namespace (and not attached):
 [1] tidyselect_1.2.0       xfun_0.39              bslib_0.4.2           
 [4] shinyjs_2.1.0          testthat_3.1.8         colorspace_2.1-0      
 [7] vctrs_0.6.2            generics_0.1.3         htmltools_0.5.5       
[10] yaml_2.3.7             utf8_1.2.3             rlang_1.1.1           
[13] later_1.3.1            pillar_1.9.0           jquerylib_0.1.4       
[16] withr_2.5.0            glue_1.6.2             lifecycle_1.0.3       
[19] teal.logger_0.1.1.9004 munsell_0.5.0          gtable_0.3.3          
[22] ragg_1.2.5             fontawesome_0.5.1      teal.widgets_0.3.0    
[25] memoise_2.0.1          evaluate_0.21          labeling_0.4.2        
[28] knitr_1.43             fastmap_1.1.1          httpuv_1.6.11         
[31] shinyWidgets_0.7.6     fansi_1.0.4            Rcpp_1.0.10           
[34] xtable_1.8-4           promises_1.2.0.1       backports_1.4.1       
[37] scales_1.2.1           checkmate_2.2.0        cachem_1.0.8          
[40] jsonlite_1.8.4         farver_2.1.1           systemfonts_1.0.4     
[43] mime_0.12              brio_1.1.3             textshaping_0.3.6     
[46] ggplot2_3.4.2          digest_0.6.31          formatters_0.5.0      
[49] dplyr_1.1.2            grid_4.2.3             cli_3.6.1             
[52] tools_4.2.3            logger_0.2.2           sass_0.4.6            
[55] tibble_3.2.1           pkgconfig_2.0.3        ellipsis_0.3.2        
[58] rmarkdown_2.21         rstudioapi_0.14        R6_2.5.1              
[61] compiler_4.2.3
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

**Comment by m7pr** [comment]
This was introduced by #287 which was a response to #286

**Comment by m7pr** [comment]
closed by #307
